src/server/game/Battlefield/Battlefield.cpp

```diff
@@ -402,8 +402,8 @@ void Battlefield::AskToLeaveQueue(Player* player)
 // Called in WorldSession::HandleHearthAndResurrect
 void Battlefield::PlayerAskToLeave(Player* player)
 {
-    // Player leaving Wintergrasp, trigger Hearthstone spell.
-    player->CastSpell(player, 8690, true);
+    // Player leaving Wintergrasp, teleports to Icecrown just north of Wintergrasp.
+    player->TeleportTo(571, 5728.117188f, 2714.345947f, 697.732971f, 0.0000f);
```

Comment:
Is this coming from a sniff?

Reply:
This is what i changed, actually no needs to trigger Hearthstone spell and having Cooldown, is so weird! So i changed it to Icecrown, North of Wintergrasp instead of the Hearthstone, what i did wrong?

Reply:
Because this should be sniffed. iirc triggering HS is the correct

Reply:
I am current researching the subject right now. Will update with what i find

Reply:
ok. @Alexander990608  so follow up. leaving wintergrasp sends you to dalaran, previous in 3.2 you were sent to somewhere north norther east of ice crown, and before that it just used your hearth stone
Dalaran Location is:
.go xyz 5804.1499 624.7710 647.7670 571 1.6400
![image](https://user-images.githubusercontent.com/16887899/158647076-cdd11610-0df7-4d52-9ed0-89d2ba7e4352.png)
Adjust your pr to:
    // Player leaving Wintergrasp, teleport to Dalaran.
    player->TeleportTo(571, 5804.1499f, 624.7710f, 647.7670f, 1.6400f);
    
 and remove your conf change then you will have my vote for approval and merge.

Reply:
can you sniff this? Because I am 99% sure that a spell teleports you

Reply:
I am not saying that. I just want a sniff of this behaviour because I am 99% sure the teleportation, to whereever it may lead, is by a spell and not hardcoded random teleport values

Reply:
> I am not saying that. I just want a sniff of this behaviour because I am 99% sure the teleportation, to whereever it may lead, is by a spell and not hardcoded random teleport values


Go to Wintergrasp when battle starts, then decline that you don't want to participate it or wait 20 seconds and you will be kicked off and teleports you in Icecrown, North of Wintergrasp. So it must be the same with the button. :)

Reply:
Yes I am still **99% sure** that it should be **handled with a spell** not by hardcoded **coordinates**